src/Analyzers/CSharp/Analyzers/UseCollectionExpression/CSharpUseCollectionExpressionForFluentDiagnosticAnalyzer.cs

```diff
@@ -133,46 +133,57 @@ private void AnalyzeMemberAccess(SyntaxNodeAnalysisContext context, INamedTypeSy
     {
         // Because we're recursing from top to bottom in the expression tree, we build up the matches in reverse.  Right
         // before returning them, we'll reverse them again to get the proper order.
-        using var _ = ArrayBuilder<CollectionExpressionMatch<ArgumentSyntax>>.GetInstance(out var matchesInReverse);
-        if (!AnalyzeInvocation(text, state, invocation, addMatches ? matchesInReverse : null, out var existingInitializer, cancellationToken))
+        using var _1 = ArrayBuilder<CollectionExpressionMatch<ArgumentSyntax>>.GetInstance(out var preMatchesInReverse);
+        using var _2 = ArrayBuilder<CollectionExpressionMatch<ArgumentSyntax>>.GetInstance(out var postMatchesInReverse);
```

Comment:
the general idea here is that when converting to a collection expr, there is often:

1. values that should be inserted into the collection expr *prior* to the values in any existing initializer.
2. the values from an existing initializer.  (note: the initializer is also used to help determine how to format/shape the collection, so we want to pass it along, not convert it into the matches-array)
3. values taht should be inserted into the collection expr *after* any values in the xisting initializer.

This PR updates everything to follow this model.